crates/zkvm/adapters/risc0/src/input.rs

```diff
@@ -36,9 +38,10 @@ impl<'a> ZkVmInputBuilder<'a> for Risc0ProofInputBuilder<'a> {
         Ok(self)
     }
 
-    fn write_proof(&mut self, item: strata_zkvm::AggregationInput) -> ZkVmResult<&mut Self> {
+    fn write_proof(&mut self, item: &AggregationInput) -> ZkVmResult<&mut Self> {
         // Learn more about assumption and proof compositions at https://dev.risczero.com/api/zkvm/composition
-        let receipt: Receipt = bincode::deserialize(item.proof().as_bytes())?;
+        let journal = item.receipt().public_values.as_bytes().to_vec();
+        let inner: InnerReceipt = bincode::deserialize(item.receipt().public_values.as_bytes())?;
```

Comment:
Same here about bincode.  It might be fine, not sure what is expected or what needs to be stable.

Reply:
Right now, `Proof` that we have defined actually is the serialized version of:
- `InnerReceipt` in case of Risc0
- `SP1Proof` in case of SP1

To serialize/deserialize we can use any supported serializer/deserializer. It should be consistent in the 
- `ZkVmHost::prove` - where the serialization happens
- `ZkVmHost::verify` - where they are converted back to their expected type
- `ZkVmInputBuilder::write_proof` - where the proof is written as input for the next program.